packages/mobile/src/identity/feelessVerification.ts

```
@@ -0,0 +1,1061 @@
+      ])
+
+      receiveMessageTask?.cancel()
+      if (Platform.OS === 'android') {
+        autoRetrievalTask?.cancel()
+      }
+    }
+
+    // Intentionally calling this a second time. No-op if they already registered so no harm done
+    yield call(feelessDekAndWalletRegistration, komenciKit, walletAddress)
+    yield call(endFeelessVerification)
+    return true
+  } catch (error) {
+    Logger.error(TAG, 'Error occured during feeless verification flow', error)
+    yield all([
+      call(storeTimestampIfKomenciError, error, true),
+      put(feelessSetVerificationStatus(VerificationStatus.Failed)),
+      put(showErrorOrFallback(error, ErrorMessages.VERIFICATION_FAILURE)),
+    ])
+    return error.message
+  } finally {
+    receiveMessageTask?.cancel()
+    if (Platform.OS === 'android') {
+      autoRetrievalTask?.cancel()
+    }
+  }
+}
+
+function* fetchKomenciReadiness(komenciKit: KomenciKit) {
+  Logger.debug(TAG, '@fetchKomenciReadiness', 'Starting fetch')
+  const feelessVerificationState: FeelessVerificationState = yield select(
+    feelessVerificationStateSelector
+  )
+
+  const { komenci } = feelessVerificationState
+
+  if (!features.KOMENCI) {
+    throw new KomenciDisabledError()
+  }
+
+  const serviceStatusResult: Result<true, KomenciDown> = yield call([
+    komenciKit,
+    komenciKit.checkService,
+  ])
+  if (!serviceStatusResult.ok) {
+    Logger.debug(TAG, '@fetchKomenciReadiness', 'Service down')
+    throw serviceStatusResult.error
+  }
+
+  if (hasExceededKomenciErrorQuota(komenci.errorTimestamps)) {
+    Logger.debug(TAG, '@fetchKomenciReadiness', 'Too  many errors')
+    throw new KomenciErrorQuotaExceeded()
+  }
+
+  yield put(
+    feelessUpdateVerificationState({
+      ...feelessVerificationState,
+      komenci: {
+        ...feelessVerificationState.komenci,
+        serviceAvailable: true,
+      },
+    })
+  )
+}
+
+function* fetchPhoneHashDetailsFromCache(e164Number: string) {
```

Could this function be refactored as a selector? It seems like its return value is entirely dependent on that state tree, so it seems odd that we call a function to     take state from one part of the tree to another.

---

The reason why it's not modeled as a selector is because we want it to throw an error in some instances. Of course, we could follow the selector call with a throw condition if `undefined` but that's basically all we are doing in this function anyways

---

Ok. Nothing about this looks incorrect, so not really worth debating now 👍